src/components/form/Form.ts

```diff
@@ -1,54 +1,74 @@
 import BaseBlock from '../../common/base-block';
 import template from './template.hbs';
-import { BaseBlockOptions, BaseBlockProps } from '../../common/types';
+import {BaseBlockOptions, BaseBlockProps} from '../../common/types';
 import Validator from '../../helpers/validator';
 
-type FormOptions = BaseBlockOptions & {
+export type FormOptions = BaseBlockOptions & {
     validators?: Record<string, Validator>
+    onSubmit?: (data: Record<string, any>) => any,
 }
 
-type FormProps = BaseBlockProps & {
-    validators?: Record<string, Validator>
+export type FormProps = BaseBlockProps & {
+    validators?: Record<string, Validator>,
+    cleanup?: boolean,
 }
 
-class Form extends BaseBlock {
+export class Form extends BaseBlock {
     protected _props: FormProps;
 
     protected _inputs: Record<string, HTMLInputElement>;
+    protected _labels: Record<string, HTMLLabelElement>;
 
     constructor(options: FormOptions, tag = 'form') {
         super(options, tag);
+        this.update({
+            events: {
+                submit: this._onSubmit,
+                focusin: this._validate,
+                focusout: this._validate,
+                input: this._validate,
+            },
+        });
     }
 
     protected render(): DocumentFragment {
-        const content = this.compile(template, this._props);
+        return this.compile(template, this._props);
+    }
+
+    protected compile(template: (props?: any) => string, props: BaseBlockProps): DocumentFragment {
+        const content = super.compile(template, props);
 
         this._inputs = Object.fromEntries(
             Array.from(content.querySelectorAll('input'))
                 .map((element) => [element.name, element]),
         );
+        this._labels = {};
+        Object.keys(this._inputs).forEach((name) => {
+            const label = content.querySelector(`label[for=${name}]`) as HTMLLabelElement;
```

Comment:
Если можно обойтись без явного приведения типов, лучше обойтись без явного приведения типов. Этот механизм на самом деле довольно опасный и костыльный. Тут можно написать так:
```suggestion
            const label = content.querySelector<HTMLLabelElement>(`label[for=${name}]`);
```